### DISCUSSION TITLE
Prisma Unit testing in JavaScript

### DISCUSSION THREAD

**Discussion by sarahmajeed** [prisma/prisma#19747] [Q&A] [answered]
https://www.prisma.io/docs/guides/testing/unit-testing

The current prisma unit test guide, gives setup details for typescript. However, I am using JavaScript. Turns out, changing the syntax alone does not work. 

Is there anyone who has created a prismaMock in JS?

**Comment by ludralph** [comment, marked as answer]
Hi @sarahmajeed 👋 

I have created the mock example from the unit testing guide you shared in Javascript using Jest. 

- https://github.com/ludralph/mocking_playground

You can take a look at it and see how I converted it from Typescript to Javascript. 

We also have another guide for mocking that uses Vitest instead Jest. This is a [link](https://www.prisma.io/blog/testing-series-1-8eRB5p0Y8o#development-environment) to the guide. This guide uses Typescript but you can also convert it to Javascript as well.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by sarahmajeed** [comment]
function:
```js
async function getAllUsers() {
  return new Promise(async (resolve, reject) => {
    prisma.users
      .findMany()
      .then((users) => {
        console.log(users);
        resolve(users)
      })
      .catch((error) => {
        reject({ error: 'Could not fetch users' })
      });
  });
}
```
test:
```js
it('should retrieve all the users from the database', async () => {

    const dummyUsers = [
      {
        firstName: "John",
        lastName: "Doe",
        email: "johnDoe@gmail.com",
      }
    ]

    prismaMock.users.findMany.mockResolvedValue(dummyUsers)
    await expect(getAllUsers()).resolves.toEqual(dummyUsers)
 
  })
```

Testing a simple get function but the test case returns an empty array, with the above implementation

**Reply by ludralph** [reply]
Hi @sarahmajeed, I think you need to take care with the naming in your code. for example:

`prismaMock.users.findMany.mockResolvedValue(dummyUsers)` should be `prismaMock.user.findMany.mockResolvedValue(dummyUsers)`. 

Note the singular `user` instead of `users`. Also in the `getAllUsers` function, you need to use the singular form `user` instead of `users` as a property on `prisma` object. i.e:

`prisma.users.findMany()` should be `prisma.user.findMany()`.

I have updated the code [example](https://github.com/ludralph/mocking_playground) in the repository I shared with you to reflect this. The output for running the test in the repository is:

```js
> jest

  console.log
    [
      {
        firstName: 'John',
        lastName: 'Doe',
        email: 'johnDoe@gmail.com',
        paidUser: false
      }
    ]

      at functions-without-context.js:25:19

 PASS  __test__/with-singleton.test.js
  ✓ should create new user  (2 ms)
  ✓ should update a users name  (1 ms)
  ✓ should fail if user does not accept terms (1 ms)
  ✓ should retrieve all the users from the database (10 ms)

Test Suites: 1 passed, 1 total
Tests:       4 passed, 4 total
Snapshots:   0 total
Time:        0.166 s, estimated 6 s
Ran all test suites.
```

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by sarahmajeed** [comment]


**Reply by ludralph** [reply]
That makes sense. Can you share your repository so I can figure out what the issue is?
